method handler crashing for new account and import raw key apis

vsmk98 opened the pull request:
**System Information**
```
➜  7nodes git:(geth-1.9.7) ✗ geth version
Geth
Version: 1.9.16-unstable
Git Commit: 02cea2330d6b4822b43a7fbaeacc12ddc8e8b1db
Git Commit Date: 20200626
Architecture: amd64
Protocol Versions: [65 64 63]
Go Version: go1.13.8
Operating System: darwin
GOPATH=
GOROOT=/usr/local/Cellar/go@1.13/1.13.8/libexec
```
**Expected Behaviour**
When running geth with clef, executing personal.importKey or personal.newAccount at geth js console should return appropriate error message.

Actual behaviour
Executing personal.importKey or personal.newAccount at geth js console returns method handler crashed error.
```
Welcome to the Geth JavaScript console!

instance: Geth/v1.9.16-unstable-02cea233-20200626/darwin-amd64/go1.13.8
coinbase: 0x46b8ce2e9d3e7a0eb653d574afa506087997461e
at block: 9 (Sat Jun 27 2020 09:16:01 GMT+0800 (+08))
 datadir: 
 modules: admin:1.0 debug:1.0 eth:1.0 miner:1.0 net:1.0 personal:1.0 rpc:1.0 txpool:1.0 web3:1.0

> personal.importRawKey("<<_keyValue_>>", "")
Error: method handler crashed
	at web3.js:6347:37(47)
	at web3.js:5081:62(37)
	at <eval>:1:22(5)

> personal.newAccount()
Passphrase:
Repeat passphrase:
GoError: Error: method handler crashed at web3.js:6347:37(47)
	at native
	at <eval>:1:1(2)
Steps to reproduce the behaviour
```
fixes #21266

holiman commented:
I only took a quick peek. I think a more 'correct' solution would be still keep the accountmanager 'above' details about specific backends, so instead of knowing that it has an external backend, it should simply fail these operations in case it finds that it indeed is missing a keystore backend. So where it now blindly assumes that the first thing is a keystore, it should just check it, and return an error if it has none. 

Skimmed it briefly, sorry if I misunderstood something. And thanks for looking into this!

vsmk98 commented:
Thanks @holiman. I will update the PR to reflect your suggestions.

holiman commented:
Another option would be to add importRawKey etc to the wallet interface, and make all non-keystores throw errors on that op.Then no casting is needed. 

vsmk98 commented:
@holiman, I have updated the PR in line with your comments. I have added `lock` and `unlock` to wallet interface and added a validation for key store backend for `NewAccount` and `ImportRawKey`. I saw the PR that you have created for this with the validation. Please let me know your views on this.

holiman commented:
With this approach, it seems we still don't get rid of things like `if len(s.am.Backends(keystore.KeyStoreType)) == 0 ` -- meaning that we're still "keystore-aware". Therefore, I don't think there's any point in adding to the wallet interface, but instead do what I did in #21279 - which just continues the path already taken. 

If you want to, you can change this PR to use that approach, and I'll merge it, and you get the credit for the commit.

vsmk98 commented:
Thanks @holiman. I am closing this PR as I feel the #21279 is addressing this in a simpler way and also includes the changes for `regexp` to prevent `personal.importRawKey` from winding up in the console history. Hopefully will be able to help in future.
